cpp/dolfinx/fem/discreteoperators.h

```diff
@@ -195,149 +200,185 @@ void interpolation_matrix(const FunctionSpace& V0, const FunctionSpace& V1,
   const std::size_t num_dofs_g = cmap.dim();
   std::span<const double> x_g = mesh->geometry().x();
 
+  namespace stdex = std::experimental;
+  using cmdspan2_t
+      = stdex::mdspan<const double, stdex::dextents<std::size_t, 2>>;
+  using cmdspan3_t
+      = stdex::mdspan<const double, stdex::dextents<std::size_t, 3>>;
+  using cmdspan4_t
+      = stdex::mdspan<const double, stdex::dextents<std::size_t, 4>>;
+  using mdspan3_t = stdex::mdspan<double, stdex::dextents<std::size_t, 3>>;
+
   // Evaluate coordinate map basis at reference interpolation points
-  const xt::xtensor<double, 2> X = element1->interpolation_points();
-  xt::xtensor<double, 4> phi(cmap.tabulate_shape(1, X.shape(0)));
-  cmap.tabulate(1, X, phi);
-  const xt::xtensor<double, 2> dphi
-      = xt::view(phi, xt::range(1, tdim + 1), 0, xt::all(), 0);
+  const auto [X, Xshape] = e1->interpolation_points();
+  std::array<std::size_t, 4> phi_shape = cmap.tabulate_shape(1, Xshape[0]);
+  std::vector<double> phi_b(
+      std::reduce(phi_shape.begin(), phi_shape.end(), 1, std::multiplies{}));
+  cmdspan4_t phi(phi_b.data(), phi_shape);
+  cmap.tabulate(1, X, Xshape, phi_b);
+  auto dphi
```

Comment:
Could you add a comment here saying that this is a bug, and should be done at every quadrature point? 
This should now be a lot easier with subspans, i.e. the second dimension is `p` instead of 0